ngraph/core/include/ngraph/op/constant.hpp

```diff
@@ -231,11 +232,24 @@ namespace ngraph
                 /// \param shape The shape of the tensor constant.
                 /// \param values An initializer_list of values to use as the constant data.
                 template <typename T>
-                static std::shared_ptr<op::v0::Constant>
-                    create(const element::Type& type, Shape shape, std::initializer_list<T> values)
+                static std::shared_ptr<Constant> create(const element::Type& type,
+                                                        const Shape& shape,
+                                                        std::initializer_list<T> values)
                 {
-                    auto result =
-                        std::make_shared<op::v0::Constant>(type, shape, std::vector<T>{values});
+                    auto result = std::make_shared<Constant>(type, shape, std::vector<T>{values});
+                    result->validate_and_infer_types();
```

Comment:
Actually we don't need to call validate_and_infer_types because it is already executed inside operation c-tor.